### DISCUSSION TITLE
Extension to get all Groups via mqtt

### DISCUSSION THREAD

**Discussion by Schnittcher** [Koenkk/zigbee2mqtt#9927] [General]
Hello @Koenkk,

I want to develop an extension to publish all groups via mqtt with members and exposes.
Can you help me with the code?

With this I get all the IDs and members: 
`let groups = this.zigbee.groups();`

But the names are still missing. Can you help me to map the IDs to names for my mqtt payload an get all exposes for this groups?

With:

`let groups = this.settings.getGroups();`
I get the names, but the exposes are missing, too. Can you give me a tip?

Thanks in advance,
kai

**Comment by sjorge** [comment]
https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-groups
You should be able to subscribe to `zigbee2mqtt/bridge/groups` it will publish on subscribe because retain is set for that topic, or when there is a change.

Groups don't have expose information though, I just grab it from the first member in the group. You can find the members in the previous mentioned output and then grab the expose data from the first member out of `zigbee2mqtt/bridge/devices`

So no need to write an extension for that, if you need the data inside and extension I think you can just walk the groups output and get the expose info form the first group member too.

**Comment by Schnittcher** [comment]
If I take the values from the first device, the min and max values may not be correct.

**Reply by sjorge** [reply]
Then you need to loop all members and keep the highest minimum value and the smallest maximum value to figure out the range all bulbs in the group accept. Likewise for other expose date ideally you eliminate the entry if not all devices support it.
